core/Pos/Core/Types.hs

```diff
@@ -149,19 +155,39 @@ data AddrType
     | ATUnknown !Word8
     deriving (Eq, Ord, Generic, Typeable, Show)
 
--- TODO [CSL-1489] This stake distribution is currently unused!
-
 -- | Stake distribution associated with an address.
 data AddrStakeDistribution
     = BootstrapEraDistr
     -- ^ Stake distribution for bootstrap era.
     | SingleKeyDistr !StakeholderId
     -- ^ Stake distribution stating that all stake should go to the given stakeholder.
-    | MultiKeyDistr ![(StakeholderId, Coin)]
+    | UnsafeMultiKeyDistr !(Map StakeholderId CoinPortion)
```

Comment:
It was discussed in https://issues.serokell.io/issue/CSL-1483
It's primary for UX (so that you don't need to regenerate an address when you only change output value and want to keep same portions).